conference/cfp.py

```diff
@@ -0,0 +1,419 @@
+    talks = dump_all_talks_for_conference_to_dict(current_conf)
+    return JsonResponse({'talks': talks})
+
+
+def extract_initial_speaker_data_from_user(user):
+    attendee = user.attendeeprofile
+
+    return {
+        'users_given_name': user.assopy_user.name(),
+        'is_minor': attendee.is_minor,
+        'gender': attendee.gender,
+        'job_title': attendee.job_title,
+        'bio': getattr(attendee.getBio(), "body", ""),
+        'company': attendee.company,
+        'company_homepage': attendee.company_homepage,
+        'phone': attendee.phone,
+    }
+
+
+def dump_all_talks_for_conference_to_dict(conference: Conference):
+
+    talks = Talk.objects.filter(conference=conference.code)
+    output = [
+        dump_relevant_talk_information_to_dict(talk)
+        for talk in talks
+    ]
+    return output
+
+
+def dump_relevant_talk_information_to_dict(talk: Talk):
+
+    output = {
+        'title': talk.title,
+        'uuid': talk.uuid,
+        'slug': talk.slug,
+        'type': talk.type,
+        'type_display': talk.get_type_display(),
+        'subtitle': talk.sub_title,
+        'abstract_short': talk.abstract_short,
+        'abstract': talk.getAbstract().body,
+        'abstract_extra': talk.abstract_extra,
+        'python_level': talk.get_level_display(),
+        'domain_level': talk.get_domain_level_display(),
+        "tags": [t.name for t in talk.tags.all()],
+        'speakers': [],
+    }
+
+    for speaker in talk.get_all_speakers():
+        ap = speaker.user.attendeeprofile
+        output['speakers'].append({
+            'name': speaker.user.assopy_user.name(),
+            'email': speaker.user.email,
+            'company': ap.company,
+            'company_homepage': ap.company_homepage,
+            'bio': ap.getBio().body,
+            'phone': ap.phone,
+        })
+
+    return output
+
+
+def save_information_from_speaker_form(user, cleaned_data):
+    user.first_name = cleaned_data['users_given_name']
+    user.save()
+
+    ap = user.attendeeprofile
+    ap.phone = cleaned_data['phone']
+    ap.is_minor = cleaned_data['is_minor']
+    ap.gender = cleaned_data['gender']
+    ap.job_title = cleaned_data['job_title']
+    ap.company = cleaned_data['company']
+    ap.company_homepage = cleaned_data['company_homepage']
+    ap.setBio(cleaned_data['bio'])
+    ap.save()
+
+    speaker, _ = Speaker.objects.get_or_create(user=user)
+    return speaker
+
+
+def add_speaker_to_talk(speaker, talk):
+    ts, _ = TalkSpeaker.objects.get_or_create(talk=talk, speaker=speaker)
+    return None
+
+
+class AddSpeakerToTalkForm(forms.ModelForm):
+
+    users_given_name = forms.CharField(label="Name of the speaker")
+    is_minor = forms.BooleanField(
+        label="Are you a minor?",
+        help_text=(
+            "Please select this checkbox if you're going to be under 18"
+            "years old on July 8th 2019"
+        ),
+        # required=False, because django forms... it means that unless someone
```

Comment:
I guess it gets rendered as a checbox as opposed to a radio button? I think there is a way to get radio buttons here, but it might not work with boolean fields. Not really an issue, just thinking out loud.

Reply:
yeah, checkbox is fine in this case, it's just webbrowsers/django thing that we need to pay attention to.